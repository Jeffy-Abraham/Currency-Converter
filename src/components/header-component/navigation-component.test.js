import React from "react";
import Header from "./navigation-component";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import checkPropTypes from "check-prop-types";

configure({ adapter: new Adapter() });
const setUp = (props = {}) => {
  const component = shallow(<Header {...props} />);
  return component;
};
//Check if correct prop types are accurate
describe("Nav Component", () => {
  describe("Checking PropType", () => {
    it("Do not throw a warning", () => {
      const expectedProps = {
        Links: "/CURRENCYEXCHANGE",
        activeLink: 1,
      };
      const propsErr = checkPropTypes(
        Header.propTypes,
        expectedProps,
        "props",
        Header
      );
      expect(propsErr).toBeUndefined();
    });
  });
});
//Navigation has container
describe("Navigation component", () => {
  it("it should render without errors", () => {
    const component = setUp();

    const wrapper = component.find(".container");
    expect(wrapper.length).toBe(1);
  });
});
//Find the Active Link
describe("Navigation component", () => {
  it("it should render only one active link", () => {
    const component = setUp();
    const Link = component.find(".active-link");
    expect(Link.length).toBe(1);
  });
});
describe("A snapshot is created for HeaderComponent", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
