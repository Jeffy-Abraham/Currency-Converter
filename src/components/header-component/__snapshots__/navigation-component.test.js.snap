// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A snapshot is created for HeaderComponent renders correctly 1`] = `
<div>
  <nav>
    <div
      className="container"
    >
      <a
        href="/"
        onClick={[Function]}
      >
         
        <div
          className="active-link"
          onClick={[Function]}
        >
          CURRENCY CONVERTER
        </div>
      </a>
      <a
        href="/exchange"
        onClick={[Function]}
      >
         
        <div
          className=""
          onClick={[Function]}
        >
          CURRENT EXCHANGE RATES
        </div>
      </a>
      <div
        className="animation-start-home"
      />
    </div>
  </nav>
</div>
`;
